File: tests/codegen/snapshots/python/union_with_typename.py
from typing_extensions import TypedDict
from typing import Optional, Union

class OperationNameResultUnionAnimal(TypedDict):
    # typename: Animal
    age: int

class OperationNameResultUnionPerson(TypedDict):
    # typename: Person
    name: str

OperationNameResultUnion = Union[OperationNameResultUnionAnimal, OperationNameResultUnionPerson]

class OperationNameResultOptionalUnionAnimal(TypedDict):
    # typename: Animal
    age: int

class OperationNameResultOptionalUnionPerson(TypedDict):
    # typename: Person
    name: str

OperationNameResultOptionalUnion = Union[OperationNameResultOptionalUnionAnimal, OperationNameResultOptionalUnionPerson]

class OperationNameResult(TypedDict):
    union: OperationNameResultUnion
    optional_union: Optional[OperationNameResultOptionalUnion]

File: tests/codegen/snapshots/typescript/union_with_typename.ts
type OperationNameResultUnionAnimal = {
    age: number
}

type OperationNameResultUnionPerson = {
    name: string
}

type OperationNameResultUnion = OperationNameResultUnionAnimal | OperationNameResultUnionPerson

type OperationNameResultOptionalUnionAnimal = {
    age: number
}

type OperationNameResultOptionalUnionPerson = {
    name: string
}

type OperationNameResultOptionalUnion = OperationNameResultOptionalUnionAnimal | OperationNameResultOptionalUnionPerson

type OperationNameResult = {
    __typename: string
    union: OperationNameResultUnion
    optional_union: OperationNameResultOptionalUnion | undefined
}
